main/src/use_case/delete_friend/DeleteFriendInteractor.java

```diff
@@ -0,0 +1,31 @@
+package use_case.delete_friend;
+
+import entity.User;
+
+public class DeleteFriendInteractor implements DeleteFriendInputBoundary {
+
+    final DeleteFriendUserDataAccessInterface userDataAccessInterface;
+    final DeleteFriendOutputBoundary userPresenter;
+    final User user;
+
+    public DeleteFriendInteractor(DeleteFriendUserDataAccessInterface userDataAccessInterface,
+                                  DeleteFriendOutputBoundary userPresenter, User user) {
+        this.userDataAccessInterface = userDataAccessInterface;
+        this.userPresenter = userPresenter;
+        this.user = user;
+    }
+
+    @Override
+    public void execute(DeleteFriendInputData deleteFriendInputData) {
+        User userToAdd = userDataAccessInterface.get(deleteFriendInputData.getFriendUsername());
+        if (!userDataAccessInterface.existsByName(deleteFriendInputData.getFriendUsername())) {
+            userPresenter.prepareFailView("User does not exist! Please check the username that you want to delete!");
+        } else if (!user.getFriends().contains(userToAdd)) {
+            userPresenter.prepareFailView("User does not exist in your friend list.");
+        } else {
+            user.addFriend(userToAdd);
+            DeleteFriendOutputData outputData = new DeleteFriendOutputData(true, userToAdd.getUserName());
```

Comment:
I think this line will produce an error since you are passing the delete friend outputdata which contains the wrong message.